src/main/java/de/tum/in/www1/artemis/service/user/UserService.java

```diff
@@ -483,6 +495,29 @@ public void changePassword(String currentClearTextPassword, String newPassword)
         });
     }
 
+    /**
+     * Check the username and password for validity. Throws Exception if invalid.
+     * @param username The username to check
+     * @param password The password to check
+     */
+    public void checkUsernameAndPasswordValidityElseThrow(String username, String password) {
+        if (username == null || username.length() < USERNAME_MIN_LENGTH) {
+            throw new AccessForbiddenException("The username has to be at least " + USERNAME_MIN_LENGTH + " characters long");
+        }
+        else if (username.length() > USERNAME_MAX_LENGTH) {
+            throw new AccessForbiddenException("The username has to be less than " + USERNAME_MAX_LENGTH + " characters long");
+        }
```

Comment:
Single responsibility: I suggest extracting this block and the other one below to two separate methods `checkUsernameValidityElseThrow()` and `checkPasswordValidityElseThrow()`.